pkg/scheduler/core/generic_scheduler.go

```diff
@@ -614,13 +612,15 @@ func addNominatedPods(pod *v1.Pod, meta predicates.PredicateMetadata,
 // add the nominated pods. Removal of the victims is done by SelectVictimsOnNode().
 // It removes victims from meta and NodeInfo before calling this function.
 func podFitsOnNode(
+	pluginContext *framework.PluginContext,
 	pod *v1.Pod,
 	meta predicates.PredicateMetadata,
 	info *schedulernodeinfo.NodeInfo,
 	predicateFuncs map[string]predicates.FitPredicate,
 	queue internalqueue.SchedulingQueue,
 	alwaysCheckAllPredicates bool,
-) (bool, []predicates.PredicateFailureReason, error) {
+	filterPluginsFunc func(pc *framework.PluginContext, pod *v1.Pod, nodeName string) *framework.Status,
```

Comment:
why are you using the function as a parameter? what about the framework

Reply:
Thanks @draveness 
IMO, from now on, it is sufficient to pass filterPluginsFunc, which seems to be better from the perspective of minimum function implementation and testing.

Reply:
On the other hand, this approach has the following drawbacks:
- Harder to read, IDEs can't find all the references to Framewokr.RunFilterPlugins
- We have repeated code for the function prototype. Refactoring takes longer.

If we have a proper fake Framework, testing shouldn't be a problem. Also, we should prefer to test exported interfaces.

Reply:
> If we have a proper fake Framework, testing shouldn't be a problem. Also, we should prefer to test exported interfaces.

We do have a fakeFramework in scheduling_queue_test.go

Reply:
@ahg-g Agree with you, one more parameter is really unnecessary, whether it is `framework` or `filterPluginsFunc`.